src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -77,8 +77,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+	
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    String textWithImagesReplaced = text.replaceAll(regex, replacement);
 
-    Message message = new Message(user, text);
+    Message message = new Message(user, textWithImagesReplaced);
```

Comment:
Out of curiosity, why did you decide to replace the text on doPost, vs on doGet or on the client? Just want to get an idea of your thought process :).

Reply:
Th reason for that is when we replace the text on doPost, it is stores once and for all in the image tag format, and we don't have to worry about it after. In contrast, if we do it on doGet or client side, it seems like extra work to replace the text every time the user login to their page (considering that each user might have multiple images posted) . I couldn't think of a reason for now that requires us to keep the original format of the user's message. Even if the user want to edit their message later( they cannot edit any messages at this point), they can copy and paste a new url  and when they click "save changes" button, I assumed it would be a doPost method that can replace the text and save the edited message in place of the old message in Datastore. I still don't know how the edit message function works, but I think it images would still be editable even if we replace the text on doPost and save it to Datastore.

Reply:
I really like to get a different perspective of this, so I would appreciate it if you give me more insight about what would be advantages/ or disadvantages of these three options :)